### PR TITLE
RN: Add error handler on WS to fix crashing on page reload

### PR DESCRIPTION
**Pull request by Hypnosphi**
Issue: https://github.com/storybooks/storybook/issues/2923

## What I did
Added error handling according to https://github.com/websockets/ws/issues/1256#issuecomment-364988689

## How to test
Launch CRNA example, refresh desktop browser tab (it should be Chrome)

### CODE REVIEW

File: lib/channel-websocket/src/index.js [hunk 1]
```diff
@@ -52,6 +52,8 @@ export class WebsocketTransport {
       this._handler(event);
     };
     this._socket.onerror = e => {
+      // Ignore network errors like `ECONNRESET`, `EPIPE`, etc.
+      if (e.errno) return;
```
Comment (lpinca) [inline lib/channel-websocket/src/index.js]: This will also catch connection errors like `ECONNREFUSED`, not sure if wanted.

Comment (Hypnosphi) [reply lib/channel-websocket/src/index.js]: What would be the proper filter to only catch crome tab reload?

Comment (lpinca) [reply lib/channel-websocket/src/index.js]: Is `WebSocket` here the browser native `WebSocket` object or `ws` or it depends on the context?

Comment (lpinca) [reply lib/channel-websocket/src/index.js]: I think this is the browser `WebSocket` as per https://github.com/storybooks/storybook/pull/3002/files#diff-432fa23ac718b94ed5be1b2802371d41R3. In this case you don't need this, the fix on the server is sufficient and correct.

**Comment by danielduan** [comment]
saw you denied the two screenshot changes. is that because of the async nature of screenshots or is this PR not ready to go yet?

**Review by danielduan — APPROVED** [review]

**Comment by danielduan** [comment]
@tmeasday does the screenshots happen when `document.ready` or is there some sort of timer that gets triggered? I think the screenshots here look like they're taken too early.

There was one I just fixed in my other PR where the early blank screenshot was set as the correct test set. I don't think it's a huge issue but maybe your other customers might mind if the timing is off.

**Comment by tmeasday** [comment]
@danielduan in this case it is due to the stories rendering an iframe, which we then need to wait to load. You can read more detail about what's happening here: https://github.com/storybooks/storybook/pull/2958

**Comment by nikhedonia** [comment]
could you someone publish 3.3.14 to npm? - many thanks

**Comment by Hypnosphi** [comment]
It's actually published:
https://www.npmjs.com/package/@storybook/react-native

> 3.3.14 is the latest of 73 releases

**Pull request merged**